packages/console/src/components/Button/index.tsx

```diff
@@ -32,9 +36,33 @@ const Button = ({
   title,
   icon,
   className,
+  isLoading = false,
+  loadingDelay = 500,
+  onClick,
   ...rest
 }: Props) => {
   const { t } = useTranslation();
+  const [showSpinner, setShowSpinner] = useState(false);
+
+  useEffect(() => {
+    // eslint-disable-next-line @silverhand/fp/no-let
+    let timeout = 0;
+
+    // Delay showing the spinner after 'loadingDelay' milliseconds
+    if (isLoading) {
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      timeout = setTimeout(() => {
+        setShowSpinner(true);
+      }, loadingDelay);
+    } else {
+      clearTimeout(timeout);
+      setShowSpinner(false);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [isLoading, loadingDelay]);
```

Comment:
nit: feels like could extract to a useBounce hook, and use a `useRef` to store the timeout instance.

Reply:
I thought about that but this is slightly different from a classic `debounce` use case.
When `isLoading` turns true, we need to delay the loading effect in 500ms, but when `isLoading` turns false, we need to immediately stop the loading.
So if we use a debounce hook in here, I suspect the `isLoading => false` would probably also be delayed, which is not what I want.

Reply:
Btw, I have updated the code to use `ref` instead of `let`